Ramki mod (#5)
nedmsmith/draft-klspa-wimse-verifiable-geo-fence: draft-lkspa-wimse-verifiable-geo-fence.md

* WIM gives signed Workload ID (WID) with geographic boundary as an additional field. This could be a certificate or a token.

# System boot/reboot and Agent start/restart time attestation/remote verification
The location agent, which is a modified SPIFFE/SPIRE (spire) agent using a geo-location plugin mechanism, is a daemon running on bare-metal Linux OS Host (H) as a process with direct access to TPM (root permissions for TPM 2.0 access may be needed for certain Linux distributions for certain H hardware configurations). The agent can gather the location from host local location sensors (e.g. GPS, GNSS). The agent has a TPM plugin (spire-tpm) which interacts with the TPM. The server (SPIFFE/SPIRE server) is running in cluster which is isolated from the cluster in which the agent is running.

> ## Boot/reboot time attestation/remote verification of OS for integrity and proof of residency on H

Reviewer: ```suggestion
## Attestation of OS Integrity and Proof of Residency on Host
```

Remote Verification: The remote server checks the integrity of the attestation report and validates the measurements against known good values from the set of trusted hosts in the shared datastore. The server also validates that the TPM EK certificate has not been revoked and part of approved list of TPM EK identifiers associated with hardware platform. At this point, we can be sure that the hardware platform is approved for running workloads and is running an approved OS.

> ## Start/restart time attestation/remote verification of agent for integrity and proof of residency on H

Reviewer: ```suggestion
## Start/restart time attestation/remote verification of agent for integrity and proof of residency on Host
```

How is this section different from the previous section that is almost the same name?

* Server verifies that the decrypted secret is the same it used to build the challenge

> * Server creates a SPIFFE ID along with the sha256sum of the TPM AK public key. Server stores agent SPIFFE ID mapping to TPM AK public key in a shared datastore.

Reviewer: ```suggestion
* Server creates a SPIFFE ID along with the sha256sum of the TPM AK public key.
Server stores agent SPIFFE ID mapping to TPM AK public key in a shared datastore.
```

* Agent is returned the attested composite location over the secure connection. Agent signs the attested composite location using TPM AK establishing proof of residency of composite location to H. This is called attested proof-of-residency aware composite location (APL).

## Step 2: Agent gets attested geographic boundary using Geo-fencing (GF) service

> * Geo-fence policies are of four flavours - (1)boolean membership of given boundary (rectangular, circular, state etc.), (2)precise location, (3)precise bounding box/circle of location, (4)approximate location with no definition of boundary. Thew first one, boolean membership of given boundary, is the most common and will be assumed as the default. They are available in the form of pre-defined templates or can be configured on demand. Enterprises, who are the user of the hosts, choose the geo-fence policies to be enforced for various hosts. Note that the hosts must belong to the set of trusted hosts in a shared ledger. The geo-fence policies applied to the set of trusted hosts are recorded in a shared ledger.

Reviewer: ```suggestion
* Geo-fence policies are of four flavours - (1) boolean membership of given boundary (rectangular, circular, state etc.), (2) precise location, (3) precise bounding box/circle of location, (4) approximate location with no definition of boundary.
The first one, boolean membership of given boundary, is the most common and will be assumed as the default. They are available in the form of pre-defined templates or can be configured on demand.
Enterprises, who are the user of the hosts, choose the geo-fence policies to be enforced for various hosts.
Note that the hosts must belong to the set of trusted hosts in a shared ledger.
The geo-fence policies applied to the set of trusted hosts are recorded in a shared ledger.
```

Reviewer: Note (1), (2), etc... won't render as numbered bullets. use:
"1. my first bullet"
"1. my second bullet"
"1. my third bullet"
without the quote marks.
